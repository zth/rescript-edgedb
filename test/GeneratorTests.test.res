open TestFramework

module Errors = {
  let unexpectedTokenWithHint = `Unexpected token: <Token IDENT "actors">
   |
 6 |       actors: {
   |       ^^^^^^
Hint: It appears that a ',' is missing in a shape before 'actors'`

  let typeOrAliasDoesNotExist = `object type or alias 'default::a' does not exist
    |
 11 |     } order by a.title
    |                ^
Hint: did you mean '.id'?`

  let objectTypeHasNoProp = `object type 'default::Movie' has no link or property 'TITLE'
    |
 11 |     } order by .TITLE
    |                ^^^^^^
`

  let unexpected = `Unexpected 'orderaaaaa'
    |
 11 |     } orderaaaaa
    |       ^^^^^^^^^^
`
}

describe("extracting errors", () => {
  test("Extracting errors #unexpectedWithTokenHint", () => {
    let error = EdgeDbGenerator__Utils.Errors.extractFromString(
      Errors.unexpectedTokenWithHint,
      ~startLoc={line: 10, col: 12},
    )
    expect(error)->Expect.toMatchSnapshot
  })

  test("Extracting errors #typeOrAliasDoesNotExist", () => {
    let error = EdgeDbGenerator__Utils.Errors.extractFromString(
      Errors.typeOrAliasDoesNotExist,
      ~startLoc={line: 10, col: 12},
    )
    expect(error)->Expect.toMatchSnapshot
  })

  test("Extracting errors #objectTypeHasNoProp", () => {
    let error = EdgeDbGenerator__Utils.Errors.extractFromString(
      Errors.objectTypeHasNoProp,
      ~startLoc={line: 10, col: 12},
    )
    expect(error)->Expect.toMatchSnapshot
  })

  test("Extracting errors #unexpected", () => {
    let error = EdgeDbGenerator__Utils.Errors.extractFromString(
      Errors.unexpected,
      ~startLoc={line: 10, col: 12},
    )
    expect(error)->Expect.toMatchSnapshot
  })
})
