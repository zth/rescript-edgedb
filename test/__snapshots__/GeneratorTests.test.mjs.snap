// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`extracting errors Extracting errors #unexpectedWithTokenHint 1`] = `
{
  "end": {
    "col": 12,
    "line": 15,
  },
  "start": {
    "col": 6,
    "line": 15,
  },
  "text": 
"Unexpected token: <Token IDENT "actors">
Hint: It appears that a ',' is missing in a shape before 'actors'"
,
}
`;

exports[`extracting errors Extracting errors #typeOrAliasDoesNotExist 1`] = `
{
  "end": {
    "col": 16,
    "line": 20,
  },
  "start": {
    "col": 15,
    "line": 20,
  },
  "text": 
"object type or alias 'default::a' does not exist
Hint: did you mean '.id'?"
,
}
`;

exports[`extracting errors Extracting errors #objectTypeHasNoProp 1`] = `
{
  "end": {
    "col": 21,
    "line": 20,
  },
  "start": {
    "col": 15,
    "line": 20,
  },
  "text": "object type 'default::Movie' has no link or property 'TITLE'",
}
`;

exports[`extracting errors Extracting errors #unexpected 1`] = `
{
  "end": {
    "col": 16,
    "line": 20,
  },
  "start": {
    "col": 6,
    "line": 20,
  },
  "text": "Unexpected 'orderaaaaa'",
}
`;
